### PR TITLE
Add z/OSMF create/update system variables API method

### LINKED ISSUE #504: Create or update system variables
Provide a system variable API method to create or update z/OS system variables. 

The implementation should support the z/OSMF REST API details and expose the required request/response handling through the SDK.

IBM REST API reference:
https://www.ibm.com/docs/en/zos/3.2.0?topic=services-create-update-system-variables

### PR DESCRIPTION
**Pull request by Muhammad7839**
Closes #504
Implements VariableCreate for create or update of z/OS system variables, 
following the same pattern as VariableDelete and VariableExport in this 
package.
An empty variables list still succeeds with no changes made, matching how 
VariableDelete already handles that case.
Note: PR #509 proposed a SystemVariable model for the get side, with 
Jackson annotations. Mine is plain, no Jackson, since this endpoint has 
no response body to parse.
Build passes locally, all tests green.

### CODE REVIEW

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    /**
+     * SystemVariable constructor.
+     *
+     * @param name        variable name
+     * @param value       variable value
+     * @param description variable description
+     * @author Muhammad Imran
+     */
+    public SystemVariable(final String name, final String value, final String description) {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java]: use

@JsonCreator
    public SystemVariable(
            @JsonProperty("name") final String name,
            @JsonProperty("value") final String value,
            @JsonProperty("description") final String description) {

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Retrieve description specified.
+     *
+     * @return description value
+     * @author Muhammad Imran
+     */
+    public String getDescription() {
+        return description;
+    }
+
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java:100]: add

    @Override
    public boolean equals(final Object obj) {
        if (this == obj) {
            return true;
        }
        if (!(obj instanceof SystemVariable)) {
            return false;
        }
        final SystemVariable other = (SystemVariable) obj;
        return Objects.equals(name, other.name)
                && Objects.equals(value, other.value)
                && Objects.equals(description, other.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, value, description);
    }

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * @version 7.0
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"name", "value", "description"})
+public final class SystemVariable {
+
+    private final String name;
+    private final String value;
+    private final String description;
+
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java:41]: add java doc for each field. and mention for name and value are required field.. description is optional.

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+     * SystemVariable constructor.
+     *
+     * @param name        variable name
+     * @param value       variable value
+     * @param description variable description
+     * @author Muhammad Imran
+     */
+    public SystemVariable(final String name, final String value, final String description) {
+        this.name = name;
+        this.value = value;
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java:56]: add ValidateUtils.checkIllegalParameter

for name and value

add unit tests to test empty and null name and value fails

**Review by frankgiordano — CHANGES_REQUESTED** [review]

File: src/main/java/zowe/client/sdk/zosvariables/methods/VariableCreate.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                EncodeUtils.encodeURIComponent(systemName);
+
+        try {
+            request.setBody(OBJECT_MAPPER.writeValueAsString(Map.of("system-variable-list", variables)));
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("error serializing variables", e);
+        }
+
+        request.setUrl(url);
+        return request.executeRequest();
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/methods/VariableCreate.java:107]: Parse the response and send a List of SystemVariables to the caller instead of the Response object. 

Example JSON response to parse.  

{"system-variable-list": [
{"name" : "var1","value":"value1","description":"description of the variable"},
{"name" : "var2","value":"value2","description":"description of the variable"}]}

Add unit tests that make sure the response is parsed properly.

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    @JsonCreator
+    public SystemVariable(
+            @JsonProperty("name") final String name,
+            @JsonProperty("value") final String value,
+            @JsonProperty("description") final String description) {
+        ValidateUtils.checkIllegalParameter(name, "name");
+        ValidateUtils.checkIllegalParameter(value, "value");
+        this.name = name;
+        this.value = value;
+        this.description = description;
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java:57]: look at isEmpty() method example I used in the project for other JSON parse object and follow the same here

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+     * Return a hash code value for this SystemVariable.
+     *
+     * @return hash code value
+     * @author Muhammad Imran
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, value, description);
+    }
+
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java]: In hindsight remove the equals and hasCode methods for now.

**Review by frankgiordano — CHANGES_REQUESTED** [review]

**Comment by Muhammad7839** [comment]
Two questions:
1. I looked for isEmpty() everywhere but couldn't find it anywhere in the current code. Turns out it used to exist in the old Response.java back in 2021, but got removed in the reorg that same year. What should I use instead now?
2. For the response parsing thing, I checked the IBM docs and this endpoint actually returns 204 with no content on success. The JSON example you sent matches their sample request body, not a response. Did you mean I should just return the list I already have back to the caller, instead of trying to parse a response that isn't there?

**Comment by frankgiordano** [comment]
typo, search for orEmpty

leave the return as respone.

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java [hunk 8]
```diff
@@ -55,11 +52,9 @@ public SystemVariable(
             @JsonProperty("name") final String name,
             @JsonProperty("value") final String value,
             @JsonProperty("description") final String description) {
-        ValidateUtils.checkIllegalParameter(name, "name");
-        ValidateUtils.checkIllegalParameter(value, "value");
-        this.name = name;
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java]: still make name and value required use orEmpty on description only

**Review by frankgiordano — CHANGES_REQUESTED** [review]

**Comment by frankgiordano** [comment]
BTW the latest changes broke the build.

File: src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java (hunk 8 above)
Comment (frankgiordano) [reply src/main/java/zowe/client/sdk/zosvariables/model/SystemVariable.java]: my bad lets remove orEmpty entirely not needed here.

**Review by frankgiordano — APPROVED** [review]
